internal/services/v1/permissions.go

```diff
@@ -476,80 +476,146 @@ func (ps *permissionServer) LookupSubjects(req *v1.LookupSubjectsRequest, resp v
 	}
 	usagemetrics.SetInContext(ctx, respMetadata)
 
-	stream := dispatchpkg.NewHandlingDispatchStream(ctx, func(result *dispatch.DispatchLookupSubjectsResponse) error {
-		foundSubjects, ok := result.FoundSubjectsByResourceId[req.Resource.ObjectId]
-		if !ok {
-			return fmt.Errorf("missing resource ID in returned LS")
+	var currentCursor *dispatch.Cursor
+	remainingConcreteLimit := 0
+
+	lsRequestHash, err := computeLSRequestHash(req)
+	if err != nil {
+		return ps.rewriteError(ctx, err)
+	}
+
+	if req.OptionalCursor != nil {
+		decodedCursor, err := cursor.DecodeToDispatchCursor(req.OptionalCursor, lsRequestHash)
+		if err != nil {
+			return ps.rewriteError(ctx, err)
 		}
+		currentCursor = decodedCursor
+	}
+
+	if req.OptionalConcreteLimit > 0 {
+		remainingConcreteLimit = int(req.OptionalConcreteLimit)
+	}
 
-		for _, foundSubject := range foundSubjects.FoundSubjects {
-			excludedSubjectIDs := make([]string, 0, len(foundSubject.ExcludedSubjects))
-			for _, excludedSubject := range foundSubject.ExcludedSubjects {
-				excludedSubjectIDs = append(excludedSubjectIDs, excludedSubject.SubjectId)
+	for {
```

Comment:
Because the limit is a *concrete* limit, and we may get back wildcards as well, so we loop until we're sure we've exhausted the results set

Reply:
The handling stream should handle all the elements up to the concrete limit, wouldn't it? From looking at the code, it would seem the only sole purpose is making sure the wildcard is sent, since in the worst-case scenario we stream up to the concrete limit, and we have to handle the potential +1 of the wildcard. Is that correct?